# Final Report
#### Wei Kong
#### Predicting Commercial Success in the Movie Industry

## Executive Summary:
<!--TProvide the decision to be impacted, Movitation, Business value.-->
We all know that movie industry is risky, but it can also bring enormous profit at the same time. That's why countless companies and people have been attracted to this industry. However, how risky is it? We all saw super hits like Titanic, and we all have seen big productions fail. There hasn't been an universal formula to make a popular movie, but it is always to our interest to choose the right movie to invest in. Personaly, I would also like to find out how to prevent the emergence of bad movies: that is how to identify them before they even started shooting.

In this analysis, I aim to produce a study to figure out specific reasons that lead to the success of a movie economically, and give advices for companies or individuals who want to enter this industry. In other words, I decided to explore how to select a movie that will be successful economically to invest in for companies and individuls who want to make a profit in this field.

## Data description/preprocessing

For the first half semester, I have been using "The Movies Dataset" on Kaggle (https://www.kaggle.com/datasets/rounakbanik/the-movies-dataset?select=movies_metadata.csv). These files contain metadata for all 45,000 movies listed in the Full MovieLens Dataset. The dataset consists of movies released on or before July 2017. Data points include cast, crew, plot keywords, budget, revenue, posters, release dates, languages, production companies, countries, TMDB vote counts and vote averages. This dataset also has files containing 26 million ratings from 270,000 users for all 45,000 movies. 

However, this dataset consists of large amount of missing entires for budget and revenue. There were only around 6000 entries with both of these features, and a large amount of data with problem, new data was chosen for the study.

### New Dataset
The new dataset was gathered from The Numbers (https://www.the-numbers.com/), a website operated by Nash Information Services, LLC., consisting of detailed information (especially on the business aspect) about various movies. Code to get and assemble the dataset were recorded in file "scraping budget and revenue", "scraping info" and "Scraping more" in the github repository (https://github.com/kongwei981126/Movie-Industry/tree/Wei)

### Overview on the dataset

In [16]:
#Preparations
#Import the libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from tqdm import tqdm
import re
import warnings


#Settings
pd.set_option('display.max_columns', None)
pd.set_option('display.float_format',  '{:,.2f}'.format)
pd.set_option('display.precision', 2)

#import the dataset
df = pd.read_csv('metadata.csv', low_memory=False, thousands=',')

#some more information
print('Rows: ' + str(df.shape[0]))
print('Colums: ' + str(df.shape[1]))
print('---------Info of dataset--------------')
print(df.info())
# print('---------Number of missing values----------')
# print(df.isnull().sum())

Rows: 6382
Colums: 34
---------Info of dataset--------------
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6382 entries, 0 to 6381
Data columns (total 34 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   moviename                         6382 non-null   object 
 1   Budget                            6382 non-null   int64  
 2   link                              6382 non-null   object 
 3   Domestic Box Office               6053 non-null   float64
 4   International Box Office          4533 non-null   float64
 5   Worldwide Box Office              4819 non-null   float64
 6   Domestic Release                  5967 non-null   object 
 7   Dom Year                          5967 non-null   float64
 8   International Release             2942 non-null   object 
 9   Int Year                          2942 non-null   float64
 10  Rating                            5983 non-null   object 
 11  Runtime 

In [17]:
df.head(1)

moviename     Budget  \
0  The Super Mario Bros. Movie  100000000   

                                            link  Domestic Box Office  \
0  Super-Mario-Bros-Movie-The-(2022)#tab=summary       260,823,700.00   

   International Box Office  Worldwide Box Office Domestic Release  Dom Year  \
0            248,398,143.00        509,221,843.00  April 5th, 2023  2,023.00   

  International Release  Int Year Rating  Runtime Franchise  \
0       April 4th, 2023  2,023.00     PG    92.00       NaN   

                                            Keywords   Genre  \
0  ['Ensemble', 'Prince/Princess', 'Royalty', 'Go...  Action   

   Production Method Creative Type  \
0  Digital Animation  Kids Fiction   

                      Production/Financing Companies  \
0  ['Universal Pictures', 'Nintendo', 'Illuminati...   

           Production Country    Languages  \
0  ['Japan', 'United States']  ['English']   

                                        Leading Cast  \
0  ['Chris Pratt', 'Anya Taylor-Joy', 'Charlie Da...   

                                     Supporting Cast  \
0  ['Keegan-Michael Key', 'Seth Rogen', 'Fred Arm...   

                    Production and Technical Credits  \
0  ['Aaron Horvath', 'Michael Jelenic', 'Matthew ...   

                               Director  Est. Domestic DVD Sales  \
0  ['Aaron Horvath', 'Michael Jelenic']                      NaN   

   Est. Domestic Blu-ray Sales  Total Est. Domestic Video Sales  \
0                          NaN                              NaN   

  Lead Ensemble Members Cameos Uncategorized Crew Narrator(s) Extras  \
0                   NaN    NaN                NaN         NaN    NaN   

  Documentary Subject(s) Interviewee(s)  
0                    NaN            NaN

In this dataset, we have 6382 observations across 34 features. However, some of them need futher processing before any other action can be taken. 



TODO    Provide the source of the dataset and how the dataset help support the development of the analytics to impact your decision(s). Also the process of data cleaning, outlier detection.

## Modeling approach

TODO    Include descriptive, predictive, and prescriptive methods employed as it applied to fulfill objectives (consider elaborating on methods used for variable & model selection, cross validation, performance metrics).

## Result and insights-

TODO   Results from your analytical work and how the results help you to fulfill the objectives

## Conclusions

TODO  Include the issue you have raised in the project, demonstrate the importance of your ideas, and to synthesize your thoughts.

## References
TODO